routers/common/middleware.go

```diff
@@ -86,19 +86,40 @@ func stripSlashesMiddleware(next http.Handler) http.Handler {
 			urlPath = rctx.RoutePath
 		}
 
-		sanitizedPath := &strings.Builder{}
-		prevWasSlash := false
-		for _, chr := range strings.TrimRight(urlPath, "/") {
-			if chr != '/' || !prevWasSlash {
-				sanitizedPath.WriteRune(chr)
+		normalizedPath := strings.TrimRight(urlPath, "/")
+		if strings.Contains(normalizedPath, "//") {
```

Comment:
Which case is this?
There's a schema (`https://…`)?
Or when is that the case?
At best add a comment explaining it.

Reply:
```suggestion
		// the following code block is a slow-path for replacing all repeated slashes "//" to one single "/"
		// if the path doesn't have repeated slashes, then no need to execute it
		if strings.Contains(normalizedPath, "//") {
```